snafu-derive/src/lib.rs

```diff
@@ -1296,7 +1297,11 @@ impl<'a> quote::ToTokens for ContextSelector<'a> {
         let backtrace_field = match *backtrace_field {
             Some(ref field) => {
                 let name = &field.name;
-                quote! { #name: std::default::Default::default(), }
+                if cfg!(feature = "std-backtraces") {
+                    quote! { #name: snafu::Backtrace::capture(), }
```

Comment:
Should we use force_capture or capture? Just using capture means the backtrace will only be created if RUST_BACKTRACE=1 at runtime. See #84

Reply:
Yeah, this is something I'd struggled a bit with when thinking through the problem. I think that using `capture` is the right place to start; we might eventually add something like `#[snafu(backtrace(force))]` to allow opting to the other function.

Reply:
I just left a comment on this in another repo, but you could have it interpret `Backtrace` as meaning `force_capture` and `Option<Backtrace>` as a regular `capture`